test/hooks.js

```diff
@@ -179,3 +179,45 @@ test('Should not hijack reply for a normal http request in the internal onError
     })
   })
 })
+
+test('Should run async hooks and still deliver quickly sent messages', (t) => {
+  t.plan(3)
+  const fastify = Fastify()
+
+  t.tearDown(() => fastify.close())
+
+  fastify.register(fastifyWebsocket)
+
+  fastify.addHook(
+    'preValidation',
+    async () => await new Promise((resolve) => setTimeout(resolve, 25))
```

Comment:
This tests that messages that arrive while async hooks are processing are still seen in the route handler